### PR TITLE
feat: add saved cards setting

### PR DESCRIPTION
**Pull request by frosso**
Fixes https://github.com/Automattic/woocommerce-payments/issues/2417

#### Changes proposed in this Pull Request

Adding the "Saved cards" option to the settings.

#### Testing instructions

- Ensure you have the "Enable grouped settings" flag enabled
- Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- The "Enable payments via saved cards" checkbox is present under the "Transaction preferences" section
- The checkbox's status is persisted on page refresh
- Updating the value updates the (pre-existing) `saved_cards` option in the gateway

-------------------

- [ ] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

**Comment by harriswong** [comment]
👀

File: includes/admin/class-wc-rest-payments-settings-controller.php [hunk 1]
```diff
@@ -186,6 +191,7 @@ public function get_settings(): WP_REST_Response {
 				'payment_request_button_size'       => $this->wcpay_gateway->get_option( 'payment_request_button_size' ),
 				'payment_request_button_type'       => $this->wcpay_gateway->get_option( 'payment_request_button_type' ),
 				'payment_request_button_theme'      => $this->wcpay_gateway->get_option( 'payment_request_button_theme' ),
+				'is_saved_cards_enabled'            => 'yes' === $this->wcpay_gateway->get_option( 'saved_cards' ),
```
Comment (harriswong) [inline includes/admin/class-wc-rest-payments-settings-controller.php:194]: You can use `$this->is_saved_cards_enabled()`. It is being used by other places in the code. You can also argue that less wrapper function is better. :)

Comment (frosso) [reply includes/admin/class-wc-rest-payments-settings-controller.php:194]: Thanks for pointing that out, I didn't notice the method - updating 👍 
```suggestion
				'is_saved_cards_enabled'            => $this->wcpay_gateway->is_saved_cards_enabled(),
```

**Review by harriswong — APPROVED** [review]
Tested between toggling the feature flag `_wcpay_feature_grouped_settings` on/off to make sure the "saved card" settings persist through. Also double checked REST payload to make sure `is_saved_cards_enabled` is passing true/false around.

LGTM

**Pull request merged**